app/helpers/devise_helper.rb

```diff
@@ -21,8 +21,13 @@ def devise_error_messages!
       flash.clear
     end
 
-    return "" if resource.errors.empty? && flash_alerts.empty?
-    errors = resource.errors.empty? ? flash_alerts : resource.errors.full_messages
+    return '' if resource.errors.empty? && flash_alerts.empty?
+
+    errors = if resource.errors.empty?
+               flash_alerts
+             else
+               resource.errors.messages.map { |key, msg| "#{I18n.t("devise.errors.keys.#{key}")} #{msg.first}" }
+             end
```

Comment:
Previously, we are using full_messages. This works well for some errors, but it automatically joins the error key (ex: :login) to the message.

As a user, I found it confusing that:
1. The field label is User Name, and yet the error message says 'Login has already been taken'. 
2. The error messages will not take into account translation of the fields/keys

This line intends to decouple the key and msg and run them through translations to give more control on the error message shown which hopefully provides better account creation experience